### PR TITLE
Some rewrites and clarifications

### PR DESCRIPTION
**Pull request by ianmiell**

### CODE REVIEW

File: docs/builder_image.md [hunk 1]
```diff
@@ -10,32 +11,30 @@ of the two basic requirements for the best possible sti performance, these are:
 
 # Required image contents
 
-The build process consists of three fundamental elements which are combined into
-final docker image, the three are: sources, scripts and builder image. During the
+The build process consists of three fundamental elements which are combined into the
+final docker image. These are: source code, sti scripts, and the builder image. Duringthe
```
Comment (bparees) [inline docs/builder_image.md]: lost a space in "Duringthe"

File: docs/builder_image.md [hunk 2]
```diff
@@ -10,32 +11,30 @@ of the two basic requirements for the best possible sti performance, these are:
...
+
+The following diagram illustrates the build workflow:
 
 ![sti workflow](./sti-flow.png "sti workflow")
 
-\* Run build's responsibility is to untar the sources, scripts and artifacts (if such
-exist) and invoke `assemble` script. If this is second run (after catching `tar`/`/bin/sh`
-error) it's responsible only for invoking `assemble` script, since both scripts and
-sources are already there.
+\* Run build's responsibility is to untar the sources, scripts and (opionally) artifacts
```
Comment (bparees) [inline docs/builder_image.md]: optionally.

**Comment by bparees** [comment]
couple minor nits but mostly this is great stuff, thanks!
@soltysh fyi.

**Comment by soltysh** [comment]
@ianmiell thanks a lot for doing this! LGTM

**Comment by bparees** [comment]
@ianmiell can you clean up the items i commented on and rebase?  not sure how bad the merge conflict is.

**Comment by bparees** [comment]
@ianmiell we've cherry-picked your changes over to a new PR so we can make the final edits.  Thanks for the contribution!
new PR w/ your original commit is here:
https://github.com/openshift/source-to-image/pull/223

**Pull request closed without merging**